### DISCUSSION TITLE
Test for the MphysAeroStruct in DAFoam 3.0.1 version failed

### DISCUSSION THREAD

**Discussion by shaheidula** [mdolab/dafoam#344] [Installation]
Dear Prof. Ping, 
I have tried to installed the new version but the regression test of the MphysAeroStruct failed. The log file is attached here. Could you please have a look?  
Thanks a lot,
[log.txt](https://github.com/mdolab/dafoam/files/9496999/log.txt)

**Comment by friedenhe** [comment]
In 301, we updated the dependency versions for pyGeo, mphys, openmdao, tacs, and funtofem. Use the latest installation guide from [here](https://dafoam.github.io/mydoc_installation_source.html)

**Comment by shaheidula** [comment]
Dear Prof. Ping, 
In fact, I reinstalled the whole package but the regression test didn't work out. Therefore, I tried to use the docker image but the aero-structural optimization tutorial didn't run well.  Could you please have a look at the error below: 
--------------------------------------------------------------------------
[[12333,1],4]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:

Module: OpenFabrics (openib)
  Host: aaf846eba4d5

Another transport will be used instead, although this may result in
lower performance.

NOTE: You can disable this warning by setting the MCA parameter
btl_base_warn_component_unused to 0.
--------------------------------------------------------------------------
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
Traceback (most recent call last):
  File "runScript_AeroStruct.py", line 10, in <module>
    from mphys.solver_builders.mphys_meld import MeldBuilder
ModuleNotFoundError: No module named 'mphys.solver_builders'
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[12333,1],6]
  Exit code:    1
--------------------------------------------------------------------------
[aaf846eba4d5:00527] 7 more processes have sent help message help-mpi-btl-base.txt / btl:no-nics
[aaf846eba4d5:00527] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages

**Comment by shaheidula** [comment]
In summary, the latest version might have some problems. Could you kindly please double-check?  Thank you very much

**Comment by friedenhe** [comment]
The runScript_AeroStruct.py in the MACH_TUTORIAL_WING case was not up to date. We have just fixed it. You should be able to run it with the docker image. Not sure why your installation did not work.. You should be able to follow the exact step to compile on Ubuntu 20.04

**Comment by shaheidula** [comment]
Dear Prof. Ping, 
Thank you very much. The tutorial is working with docker image now. I am trying to reinstall the package with the source code again.

**Reply by uttamcadambi07** [reply]
I got the same error despite following the documentation of the updated modules.

**Reply by uttamcadambi07** [reply]
Can we ignore it if we are not using Aero struct optimization?

**Reply by uttamcadambi07** [reply]
Here is my error  [aerostruct_error.txt](https://github.com/mdolab/dafoam/files/9763353/aerostruct_error.txt)

I use Ubuntu 20.04 LTS on VirtualBox. 

Installed DAFoam first thing after creating the Linux virtual machine.

No errors during installation of the other packages.

**Reply by friedenhe** [reply]
Yes, you can ignore this if you are not doing  aerostruct opt

**Reply by uttamcadambi07** [reply]
Thank you very much Dr. He.

**Reply by friedenhe** [reply]
@uttamcadambi07 We have identified the problem. It was caused by the version of the nptyping package. The fix is to do this `pip install nptyping==1.4.4` and then run the test again. We have updated the installation guide to fix this issue. Let me know if it works for you.

**Reply by uttamcadambi07** [reply]
Hi Dr. He. Thank you very much. I did what you recommended but unfortunately it does not work for me. When I installed ntyping, this is the message I got 
![Screenshot from 2022-10-17 16-42-19](https://user-images.githubusercontent.com/10186475/196234551-7ce0eee6-6a55-4c58-9484-c63b449bdb07.png)

But after this, I tried doing the regression tests and they stop at AeroStruct
